lib/valkyrie/specs/shared_specs/change_set.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+RSpec.shared_examples 'a Valkyrie::ChangeSet' do |*_flags|
+  before do
+    raise 'adapter must be set with `let(:change_set)`' unless defined? change_set
+  end
+
+  subject { change_set }
+
+  it { is_expected.to respond_to :append_id }
+  it { is_expected.to respond_to :fields }
+  it { is_expected.to respond_to :fields= }
+  it { is_expected.to respond_to :multiple? }
+  it { is_expected.to respond_to :prepopulate! }
+  it { is_expected.to respond_to :required? }
+  it { is_expected.to respond_to :validate }
+  it { is_expected.to respond_to :valid? }
+
+  it "can set an append_id" do
+    change_set.append_id = Valkyrie::ID.new("test")
+    expect(change_set.append_id).to eq Valkyrie::ID.new("test")
+    expect(change_set[:append_id]).to eq Valkyrie::ID.new("test")
+  end
+
+  describe "#fields" do
+    it "returns an array" do
+      expect(change_set.fields).to be_a Hash
+    end
+  end
+
+  describe "#fields=" do
+    it "sets fields" do
+      change_set.fields = { "totally_a_field" => [] }
+      expect(change_set.fields).to eq("totally_a_field" => [])
+    end
+  end
+
+  describe "#multiple?" do
+    it "returns a boolean" do
+      expect(change_set.multiple?(change_set.fields.keys.first)).to be_in [true, false]
+    end
+  end
+
+  describe "#prepopulate!" do
+    it "doesn't make it look changed" do
+      expect(change_set).not_to be_changed
+      change_set.prepopulate!
+      expect(change_set).not_to be_changed
+    end
+  end
+
+  describe "#required?" do
+    it "returns a boolean" do
+      expect(change_set.required?(change_set.fields.keys.first)).to be_in [true, false]
```

Comment:
This requires that the given change set has at least one field - maybe we should add that to the expectation at the top?